Source/Dafny/Compilers/Compiler-python.cs

```diff
@@ -908,12 +940,24 @@ protected class ClassWriter : IClassWriter {
     }
 
     protected override ConcreteSyntaxTree EmitArraySelect(List<string> indices, Type elmtType, ConcreteSyntaxTree wr) {
-      throw new NotImplementedException();
+      Contract.Assert(indices != null && 1 <= indices.Count);
+      var w = wr.Fork();
+      if (indices.Count == 1) {
+        wr.Write("[{0}]", indices[0]);
+      }
+      return w;
     }
 
     protected override ConcreteSyntaxTree EmitArraySelect(List<Expression> indices, Type elmtType, bool inLetExprBody,
       ConcreteSyntaxTree wr, ConcreteSyntaxTree wStmts) {
-      throw new NotImplementedException();
+      Contract.Assert(indices != null && 1 <= indices.Count);  // follows from precondition
+      var w = wr.Fork();
+      foreach (var index in indices) {
+        wr.Write("[");
+        TrExpr(index, wr, inLetExprBody, wStmts);
+        wr.Write("]");
+      }
+      return w;
```

Comment:
```suggestion
      var strings = indices.Select(index => Expr(indices[0], inLetExprBody, wStmts).ToString());
      return EmitArraySelect(strings.ToList(), elmtType, wr);
```

Reply:
?

Reply:
It already works on my branch. I’d rather not have this duplication.

Reply:
Does the test in my PR works with this code? or might need to add If condition.